lib/logflare/sources.ex

```diff
@@ -355,4 +354,14 @@ defmodule Logflare.Sources do
       x -> x
     end
   end
+
+  @spec preload_for_dashboard(list(Source.t())) :: list(Source.t())
+  def preload_for_dashboard(sources) do
+    sources
+    |> Enum.map(&preload_defaults/1)
+    |> Enum.map(&preload_saved_searches/1)
+    |> Enum.map(&put_schema_field_count/1)
+    |> Enum.sort_by(& &1.name, &<=/2)
+    |> Enum.sort_by(& &1.favorite, &>=/2)
+  end
```

Comment:
we can split these into 2 different functions, one that iterates over each item and another that does the sort.

This one isn't very efficient  because it iterates over the list 5 time, one for each map and sort_by.

Would be more efficient to do a singular map (where each op is performed element-wise), then a singular sort.

also, function name doesn't match the actual operations, would be more accurate to say `prepare_for_dashboard/1` instead. In this case, the function name is not self-documenting and would be misleading.

Reply:
I thought about it and makes a lot of sense 👍 

we can also make an intermediary step where:
1.  Use Streams to compose transformations 
2. `sort_by` we should actually return a tuple of {name, favorite} so it can sort directly with multiple params but transverse only once ( https://hexdocs.pm/elixir/1.13/Enum.html#sort_by/3-examples )